packages/page-staking/src/Nomination/index.tsx

```diff
@@ -0,0 +1,380 @@
+  const [amount, setAmount] = useState<BN | undefined | null>(null);
+  const controllerBalance: Balance | null = useBalanceClear(controllerAccountId);
+  const accountBalance: Balance | null = useBalanceClear(senderId);
+  const ownStashes = useOwnStashes();
+  const { validators, validatorsLoading } = useValidators();
+  const { t } = useTranslation();
+  const destination = 2; // 2 means controller account
+  const extrinsic = (amount && controllerAccountId)
+    ? api.tx.staking.bond(controllerAccountId, amount, destination)
+    : null;
+  const existentialDeposit = api.consts.balances.existentialDeposit;
+
+  function onStatusChange() {}
+
+  function setStepsStateAction() {
+    if (controllerAlreadyBonded && !isNominated) {
+      setStepsState((prevState): string[] => {
+        const newState = [...prevState];
+        newState[2] = 'completed';
+        newState[3] = '';
+        return newState;
+      });
+      // go to last tab
+      if (currentStep === steps[2]) {
+        setCurrentStep(steps[3]);
+      }
+    }
+    if (isNominated) {
+      setStepsState((prevState): string[] => {
+        const newState = [...prevState];
+        newState[2] = 'completed';
+        newState[3] = 'completed';
+        return newState;
+      });
+      // go to last tab
+      if (currentStep === steps[2]) {
+        setCurrentStep(steps[3]);
+      }
+    }
+  }
+
+  function balanceWrapper(text: string): React.ReactNode {
+    return (
+      <strong className='label'>{text}</strong>
+    );
+  }
+
+  function goBack() {
+    const ind = steps.indexOf(currentStep);
+    setCurrentStep(steps[ind - 1]);
+  }
+
+  function goNext() {
+    const ind = steps.indexOf(currentStep);
+    if (ind < 3) {
+      setCurrentStep(steps[ind + 1]);
+    }
+  }
+
+  function disableNext() {
+    const ind = steps.indexOf(currentStep);
+    return stepsState[ind + 1] === 'disabled';
+  }
+
+  function resetControllerInfo(accountId: string | null) {
+    setControllerAlreadyBonded(false);
+    setControllerAccountId(accountId);
+  }
+
+  /**
+   * double fees for case if fees will be changed on a small count of funds
+   * @todo - compare with account balance and throw error if more
+   */
+  function setAmountToTransfer() {
+    const minAmount = new BN(0);
+    setTransferableAmount(
+      minAmount
+        .iadd(wholeFees)
+        .iadd(wholeFees)
+        .isub(controllerBalance || new BN(0))
+    );
+  }
+
+  function calculateMaxPreFilledBalance() {
+    if (accountBalance && wholeFees && !amountToBond) {
+      // double wholeFees
+      setAmountToBond(accountBalance.isub(wholeFees).isub(wholeFees).isub(existentialDeposit));
+    }
+    return 0;
+  }
+
+  function isBalanceEnough() {
+    return (accountBalance
+      && controllerBalance
+      && existentialDeposit
+      && wholeFees
+      && accountBalance.cmp(existentialDeposit) === 1
+      && controllerBalance.cmp(wholeFees) === 1)
+  }
```

Comment:
ll these functions, and the ones above, are created on each render. This is not a good approach. Additionally, when used as props, this would also mean that all child components are being re-rendered.